templates/common/gcp/files/usr-local-bin-update-dns-server.yaml

```diff
@@ -0,0 +1,22 @@
+mode: 0755
+path: "/usr/local/bin/update-dns-server"
+contents:
+  inline: |
+    #!/bin/bash
+    # This is test code
+    
+    echo "Executing update-dns-server"
+
+    if ![-f /etc/NetworkManager/conf.d]; then
+      mkdir /etc/NetworkManager/conf.d
+      touch /etc/NetworkManager/conf.d/dns-servers.conf
+    fi
+    if ![-f /etc/NetworkManager/conf.d/dns-servers.conf]; then
```

Comment:
The redirection below will create the file. You don't need to do it explicitly.

Reply:
Even more, usually you do not need `if [...] then touch` because `touch` on a file that already exists will not modify its content. It will only change its modification and access times but as long as there is no logic comparing those, it's transparent